CenterForOpenScience/ember-osf-web
# [EMB-495][Collections] Add redirection on cancel for submit and edit page

## Purpose

Currently, hitting the `cancel` button on submit/edit page clears the entire form. This PR modifies that so that when hitting `cancel` on submit page takes the user to discover page, and doing so on edit page takes the user to project overview page.

## Summary of Changes

Two actions are declared in respective controllers of `submit` and `edit` view. These actions are passed to the `collections-submission` component so that it can be consumed.

## QA Notes

Hitting `cancel` on submit page should now take user to discover page of the provider, while for the edit page it should take user to project overview page.

## Ticket

https://openscience.atlassian.net/browse/EMB-495

# Reviewer Checklist
